### DISCUSSION TITLE
atsetcavity, identity pass

### DISCUSSION THREAD

**Discussion by simoneliuzzo** [atcollab/at#557] [Matlab AT]
I try to use the legacy mode of `atsetcavity`. 
Setting the voltage to a non zero value, lets the cavity integrator to IdentityPass. See below.

```
atsetcavity(r_RFstart,50e6,0,harm); % sets RF ON to 50MV and radiation OFF.
ans{1} = 

  struct with fields:

       FamName: 'RFC'
    PassMethod: 'IdentityPass'
        Length: -8.1899e-13
         Class: 'RFCavity'
       Voltage: 12500000
     Frequency: 4.4393e+08
    HarmNumber: 135000
       TimeLag: 0
        Energy: 4.5600e+10


```
`atsetcavity(ring, VOLT, RAD, HARM) ` used to be a "single function to remember" in order to set the radiation and cavity state of the lattice. Apparently recent updates changed the output of this function.

**Comment by lfarv** [comment]
Hi @simoneliuzzo 
Can you tell what you expected ? and give more details on you lattice: how many cavities in the cell, and how many cells in the ring ?

**Comment by simoneliuzzo** [comment]
Dear @lfarv,

atsetcavity used to give me a lattice with the features requested by the input:

`new_ring = atsetcavity(r_RFstart,50e6,0,harm)
`

`new_ring` has 
* RF votlage set to 50e6 V evenly shared among cavities.
* cavities on, RFCavityPass
* radiation off (no RadPass and phase 0.0  in the cavities)

the lattice is a proposal for the FCC-ee Z lattice, it has 168 cells, 4 dogleg sections with RF cavities and 4 final focus sections.

best regards
Simone

**Comment by lfarv** [comment]
From what I see in the history, with `radflag=0` as in you example, `atsetcavity` has always left the cavity pass unchanged, back to the the initial version (Boaz Nash, 2014).  The `atradon` call appeared in 2015 (Simone Liuzzo, 2015), only when radflag in 1. Unless I missed something…

So could you be more precise on what is wrong in your example ? For voltage, frequency and harm, I miss the data !

**Comment by lfarv** [comment]
Ok. I found what I missed for radflag=0: `atradoff(ring, 'CavityPass')`: very strange, but does indeed turn on cavities. I will correct that.

**Comment by simoneliuzzo** [comment]
It looks as my mistake then. Apparently I am setting IdentityPass somewhere before. 

However I think CavityPass was set by atsetcavity for radiation off case, for example here at line 38:
https://github.com/atcollab/at/blob/e4847427d135295a7b5fb28f82912ad1db0b6b6d/atmat/atphysics/LongitudinalDynamics/atsetcavity.m

May be I am wrong. 

best regards
Simone

**Reply by lfarv** [reply]
Got it 1 mn before you ! Anything else wrong ?

**Comment by lfarv** [comment]
Simone,

You should have the corrected version in the master and exact_pass branches

Laurent
